### PR TITLE
CA-376879: VLAN PIF created in pool.join is shown as disconnected

### PR DESCRIPTION
**Pull request by minglumlu**
The issue to be fixed in this commit is that when a host joins a pool in which the management network is on a VLAN network, the new untagged PIF of the joined host shows disconnected always.

Normally, the VLAN untagged PIF should share PIF_metrics with the tagged PIF so that the PIF_metrics could be updated correctly. This sharing is ensured by 'Xapi_vlan.create'. While in pool.join, the untagged PIF is created by 'Xapi_pif.pool_introduce', in which the 'metrics' of the new untagged PIF doesn't share with the tagged PIF. This causes the 'carrier' of 'metrics' of the new untagged PIF be 'false' forever. And consequenctly the management network shows disconnected always.

Can't switch to use 'Xapi_vlan.create' simply to fix this issue as the call can't set the "management" field of the untagged PIF correctly. The 'Xapi_pif.pool_introduce' can do this, but can't set the sharing of PIF_metrics correctly. What's more, from engineering's perspective, it is not worth to change these two API calls for this minor issue.

So the fix is as simple as to reset the PIF_metrics of untagged PIF to the one used by tagged PIF when creating the VLAN object in pool.join. In this case, the only drawback is an unused PIF_metrics is created and then destroyed in pool.join.

### CODE REVIEW

File: ocaml/xapi/xapi_vlan.ml
Comment (lindig) [inline ocaml/xapi/xapi_vlan.ml:33]: Would suggest to use names `untagged_pif` and `tagged_pif` or similar - tagged vs untagged is the theme here.

**Review by lindig — APPROVED** [review]

File: ocaml/xapi/xapi_vlan.ml
Comment (psafont) [inline ocaml/xapi/xapi_vlan.ml:38]:
```suggestion
      "%s Set PIF_metrics (%s) of VLAN untagged PIF (%s) to the one (%s) of VLAN \
       tagged PIF (%s)"
      __FUNCTION__
```

**Comment by lindig** [comment]
Please squash before merging

**Review by psafont — APPROVED** [review]

**Comment by psafont** [comment]
The commit weren't squashed @minglumlu next time I won't approve until they are

**Comment by psafont** [comment]
For some reason github's UI shows the fixups as merged, but in reality only one squashed commit reached master

**Pull request merged**
